mpu6050/mpu6050.c

```diff
@@ -14,56 +16,94 @@ enum {
 	Z = 2
 };
 
+#define SAMPLES_COUNT (10)
+
 struct mpu6050_data {
-	struct i2c_client *drv_client;
 	int accel_values[3];
 	int gyro_values[3];
 	int temperature;
+	unsigned long timestamp;
 };
 
-static struct mpu6050_data mpu6050_data;
+static struct i2c_client *mpu6050_drv_client;
+struct mpu6050_data mpu6050_data_samples[SAMPLES_COUNT];
+static int mpu6050_current_sample_idx;
+static unsigned long mpu6050_last_access_time;
```

Comment:
IMO, it's better to collect all private global variables above into single static structure